nestkernel/archiving_node.cpp

```diff
@@ -163,22 +163,22 @@ nest::Archiving_Node::get_history( double t1,
   }
   else
   {
-    std::deque< histentry >::iterator runner = history_.begin();
-    while ( ( runner != history_.end() )
-      and ( t1 - runner->t_ > -1.0
+    std::deque< histentry >::reverse_iterator runner = history_.rbegin();
+    while ( ( runner != history_.rend() )
+      and ( t2 - runner->t_ < -1.0
                 * kernel().connection_manager.get_stdp_eps() ) )
```

Comment:
If I understand the code right, then only `runner->t_` changes in the comparison. Would it then not be tidier and possibly more efficient to write
```c++
const double t2_lim = t2 + kernel().connection_manager.get_stdp_eps();
while ( ... and ( runner->t_ > t2_lim ) )
  ...
```
and the same in the loop below. At the very least we should re-organize the expression to avoid the multiplication with `-1.0`.

Reply:
I also think that you need to use `<=` in both cases to ensure identical behavior as before. Consider the case where `t1 + stdp_eps == runner->t_` for some spike. Then, in the old version, the t1-loop is not executed any more, the t2-loop is entered and the access-counter for that particular spike increased. In the new version the counter would not be increased in that case, unless we have `<=`.